bin/saya/src/main.rs

```diff
@@ -23,7 +22,7 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     saya.start().await?;
 
     // Wait until Ctrl + C is pressed, then shutdown
-    ctrl_c().await?;
+    // ctrl_c().await?;
```

Comment:
not required anymore to be able to kill the program? Or it's handled by the `.start()` function?

Reply:
Yeah but we may want to kill saya. :) In this case it would be better keeping it to ensure graceful shutdown in case of manual interruption.

Reply:
It is the opposite, this line simply prevents `saya` from closing until `ctrl+c` is pressed. Now we do the same thing by simply not returning from the `start` function until all the blocks are prover (for ephemeral shard).
A graceful shutdown would have to be implemented there.